extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,242 @@
+	                                                  [&](string_t input_str, bool input_quote) {
+		                                                  auto str = input_str.GetData();
+		                                                  auto str_size = input_str.GetSize();
+		                                                  string escaped;
+		                                                  for (size_t i = 0; i < str_size; ++i) {
+			                                                  char ch = str[i];
+			                                                  switch (ch) {
+			                                                  case '&':
+				                                                  escaped += "&amp;";
+				                                                  break;
+			                                                  case '<':
+				                                                  escaped += "&lt;";
+				                                                  break;
+			                                                  case '>':
+				                                                  escaped += "&gt;";
+				                                                  break;
+			                                                  case '"':
+				                                                  if (input_quote) {
+					                                                  escaped += "&quot;";
+				                                                  } else {
+					                                                  escaped += "\"";
+				                                                  }
+				                                                  break;
+			                                                  case '\'':
+				                                                  if (input_quote) {
+					                                                  escaped += "&#x27;";
+				                                                  } else {
+					                                                  escaped += "\'";
+				                                                  }
+				                                                  break;
+			                                                  default:
+				                                                  escaped += ch;
+			                                                  }
+		                                                  }
+		                                                  return StringVector::AddString(result, escaped);
+	                                                  });
+}
+
+static void add_blob(int &code_point, string &result) {
+	int sz = 0;
+	char c[4] = {'\0', '\0', '\0', '\0'};
+	D_ASSERT(Utf8Proc::CodepointToUtf8(code_point, sz, c));
+	result += c;
+}
+
+static string ReplaceCharref(string &str) {
+	RE2::Options options;
+	options.set_log_errors(false); // disable error logging for performance
+	options.set_case_sensitive(false);
+	const RE2 charref(R"(&(#[0-9]+;?|#[xX][0-9a-fA-F]+;?|[^\t\n\f <&#;]{1,32};?))", options);
```

Comment:
Similar to below - we are compiling this regex for every call to `ReplaceCharref` which seems rather expensive. Perhaps we can store it in the local state instead (see e.g. `RegexInitLocalState`)?

Alternatively - perhaps we can avoid using a regex here entirely?